### PR TITLE
Add label selector for Node

### PR DESCRIPTION
**Pull request by kazegusuri**
Add label selector for Node and change a few tests.

### CODE REVIEW

File: pkg/registry/minion/rest.go [hunk 1]
```diff
@@ -91,7 +91,18 @@ func (rs *REST) Get(ctx api.Context, id string) (runtime.Object, error) {
 
 // List satisfies the RESTStorage interface.
 func (rs *REST) List(ctx api.Context, label, field labels.Selector) (runtime.Object, error) {
-	return rs.registry.ListMinions(ctx)
+	nodes, err := rs.registry.ListMinions(ctx)
```
Comment (nikhiljindal) [inline pkg/registry/minion/rest.go]: I see that for Watch, we pass the label selector to the registry, which in turn passes a filter function to etcd (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/registry/etcd/etcd.go#L404) which does the actual filtering.
We should be using the same pattern at both places.

**Comment by nikhiljindal** [comment]
cc @bgrant0607 @smarterclayton to comment on the right place to do the filtering.

File: pkg/registry/minion/rest.go (hunk 1 above)
Comment (bgrant0607) [inline pkg/registry/minion/rest.go]: Yes, but node/minion needs to be converted to use the generic etcd implementation, similar to pods:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/registry/generic/etcd/etcd.go#L138
```
func (r *REST) List(ctx api.Context, label, field labels.Selector) (runtime.Object, error) {
	return r.store.List(ctx, label, field)
}
```

Comment (bgrant0607) [inline pkg/registry/minion/rest.go]: @lavalamp

Comment (smarterclayton) [inline pkg/registry/minion/rest.go]: I'd be willing to defer that after this pull.

Comment (kazegusuri) [inline pkg/registry/minion/rest.go]: I'm rewriting whole minion registry by genericetcd based, so please take a look after that.

Comment (lavalamp) [inline pkg/registry/minion/rest.go]: Yes, fix after we start using the generic etcd code.

Comment (bgrant0607) [inline pkg/registry/minion/rest.go]: @kazegusuri You've started to port the minion/node registry to generic already? Or would you like this PR to be merged first?

Comment (kazegusuri) [inline pkg/registry/minion/rest.go]: @bgrant0607 I have worked on porting and will send another pull request soon.

**Comment by kazegusuri** [comment]
close this PR in favor of #5441
